interface/reports/daily_summary_report.php

```diff
@@ -0,0 +1,350 @@
+<?php
+/*
+ *  Daily Summary Report. (/interface/reports/daily_summary_report.php
+ *  
+ *
+ *  This report shows date wise numbers of the Appointments Scheduled,
+ *  New Patients, Visited patients, Total Charges, Total Co-pay and Balance amount for the selected facility.
+ * 
+ * Copyright (C) ....
+ * 
+ * LICENSE: This program is free software; you can redistribute it and/or 
+ * modify it under the terms of the GNU General Public License 
+ * as published by the Free Software Foundation; either version 3 
+ * of the License, or (at your option) any later version. 
+ * This program is distributed in the hope that it will be useful, 
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of 
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the 
+ * GNU General Public License for more details. 
+ * You should have received a copy of the GNU General Public License 
+ * along with this program. If not, see <http://opensource.org/licenses/gpl-license.php>;. 
+ * 
+ * @package OpenEMR 
+ * @author Rishabh Software
+ * @link http://www.open-emr.org 
+ *
+ */
+
+$fake_register_globals = false;
+$sanitize_all_escapes = true;
+
+require_once("../globals.php");
+require_once "$srcdir/options.inc.php";
+require_once "$srcdir/appointments.inc.php";
+
+
+$selectedFromDate = isset($_POST['form_from_date'])?$_POST['form_from_date']:date('Y-m-d'); // From date filter
+$selectedToDate = isset($_POST['form_to_date'])?$_POST['form_to_date']:date('Y-m-d');     // To date filter
+$selectedFacility = isset($_POST['form_facility'])?$_POST['form_facility']:"";  // facility filter
+ 
+$from_date = fixDate($selectedFromDate, date('Y-m-d'));
+$to_date = fixDate($selectedToDate, date('Y-m-d'));
+
+?>
+
+<html>
+    <head>
+        <?php html_header_show(); ?>
+        <link rel="stylesheet" href="<?php echo $css_header; ?>" type="text/css">
+        <script type="text/javascript" src="../../library/textformat.js"></script>
+        <script type="text/javascript" src="../../library/js/jquery.1.3.2.js"></script>
+        <script type="text/javascript">
+
+
+            var mypcc = '<?php echo $GLOBALS['phone_country_code'] ?>';
+
+            function submitForm() {
+                var fromDate = $("#form_from_date").val();
+                var toDate = $("#form_to_date").val();
+                
+                var dayDifference =  Math.floor(( Date.parse(toDate) - Date.parse(fromDate) ) / 86400000);
+                
+                if (fromDate === '') {
+                    alert("Please select From date");
+                    return false;
+                }
+                if (toDate === '') {
+                    alert("Please select To date");
+                    return false;
+                }
+                if(dayDifference > 31){
```

Comment:
1. We have used "fetchAppointments" function to count all the appointments including recurring events in the given date ranges. 

2. We have kept date range limit to one month because of not to display long period data as it is a daily summary report. After your comment, we have removed date range limit and now user can generate report for any date range.

3. Yes, it would be useful to run the daily summary report for the each provider. So we have included providers list in the report search criteria. Now user can see the daily summary data for the specific provider.

Please see the appointments_inc.php file for these changes.

https://github.com/rishabhemr/daily_summary_report/pull/1/commits/33b7b5f30e457c6b016dcf4542cef9756ed13d97